cura/Machines/QualityManager.py

```diff
@@ -259,11 +259,13 @@ def getQualityGroups(self, machine: "GlobalStack") -> Dict[str, QualityGroup]:
                 root_material_id = self._material_manager.getRootMaterialIDWithoutDiameter(root_material_id)
                 root_material_id_list.append(root_material_id)
 
-                # Also try to get the fallback material
-                material_type = extruder.material.getMetaDataEntry("material")
-                fallback_root_material_id = self._material_manager.getFallbackMaterialIdByMaterialType(material_type)
-                if fallback_root_material_id:
-                    root_material_id_list.append(fallback_root_material_id)
+                # Also try to get the fallback materials
+                fallback_ids = self._material_manager.getFallBackMaterialIdsByMaterial(extruder.material)
+
+                if fallback_ids:
+                    root_material_id_list.extend(fallback_ids)
+                root_material_id_list = list(set(root_material_id_list))  # Weed out duplicates
```

Comment:
This set transformation makes the resulting `root_material_id_list` have a nondeterministic ordering. The ordering is used down below in the inline `addNodesToCheck` function where the first material that is compatible in the current set-up is chosen, so which material is chosen is then random.

Here's a suggestion to save you some effort: https://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-whilst-preserving-order